pkgs/tools/security/mpw/default.nix

```diff
@@ -1,38 +1,44 @@
-{ stdenv, cmake, fetchFromGitHub, ncurses, libsodium, json_c }:
+{ stdenv, cmake, fetchFromGitLab
+, ncurses, libsodium, json_c }:
 
-stdenv.mkDerivation rec {
-  name = "mpw-2.6-f8043ae";
+let
+  rev = "22796663dcad81684ab24308d9db570f6781ba2c";
 
-  src = fetchFromGitHub {
-    owner = "Lyndir";
-    repo = "MasterPassword";
-    rev = "f8043ae16d73ddfb205aadd25c35cd9c5e95b228";
-    sha256 = "0hy02ri7y3sca85z3ff5i68crwav5cjd7rrdqj7jrnpp1bw4yapi";
+in stdenv.mkDerivation rec {
+  name = "mpw-${version}";
+  version = "2.6-${builtins.substring 0 8 rev}";
+
+  src = fetchFromGitLab {
+    owner  = "MasterPassword";
+    repo   = "MasterPassword";
+    sha256 = "1f2vqacgbyam1mazawrfim8zwp38gnwf5v3xkkficsfnv789g6fw";
+    inherit rev;
   };
 
-  postUnpack = ''
-    sourceRoot+=/platform-independent/cli-c
-  '';
+  sourceRoot = "./source/platform-independent/c/cli";
 
-  preConfigure = ''
-    substituteInPlace CMakeLists.txt --replace curses ncurses
-    echo ${name} > VERSION
+  postPatch = ''
+    rm build
+    sed -i '5iset(mpw_version ${version})' CMakeLists.txt
```

Comment:
Any reason not to do this via a cmake flag (`cmakeFlags = ["-Dmpw_version=${version}"]`)?

Reply:
I *believe* that was one of the things I tried (I tried many...) before resorting to the ugly ```sed``` hack to get things working.

Reply:
You were right @lheckemann - thanks!